lib/vagrant/util/downloader.rb

```diff
@@ -292,11 +292,14 @@ def execute_curl(options, subprocess_options, &data_proc)
         # show an error message.
         if result.exit_code != 0
           @logger.warn("Downloader exit code: #{result.exit_code}")
-          parts    = result.stderr.split(/\n*curl:\s+\(\d+\)\s*/, 2)
-          parts[1] ||= ""
-          raise Errors::DownloaderError,
-            code: result.exit_code,
-            message: parts[1].chomp
+          parts    = result.stderr.match(/\n*curl:\s+\(\d+\)\s*/)
```

Comment:
Can we make this matching a bit more explicit and include the message returned so it can be provided in the error? Something like:

```ruby
check = result.stderr.match(/\n*curl:\s+\((?<code>\d+)\)\s*(?<error>.*)$/)
if check && check[:code] == "416"
  ...
```